src/components/Carousel/index.tsx

```diff
@@ -0,0 +1,252 @@
+    const endPoint = navType === LEFT ? 0 : translateValueOfLastImage
+    const prevTranslateValue = navType === LEFT ? translateValueOfLastImage : 0
+    if (translateValue === endPoint) {
+      setTranslateValue(prevTranslateValue)
+    } else {
+      setTranslateValue(prev => prev + TRANSLATE_VALUE_OF_NAV_TYPE[navType])
+    }
+  }
+
+  const handleMouseDown = (e: { clientX: SetStateAction<number> }): void => {
+    setStartClientX(e.clientX)
+    setCursorOn(true)
+  }
+
+  const handleMouseUp = (e: { clientX: SetStateAction<number> }): void => {
+    setEndClientX(e.clientX)
+    setCursorOn(false)
+  }
+  const handleTouchStart: TouchEventHandler<HTMLDivElement> = e => {
+    setEndClientX(e.touches[0].clientX)
+    setCursorOn(true)
+  }
+  const handleTouchStartEnd: TouchEventHandler<HTMLDivElement> = e => {
+    setStartClientX(e.changedTouches[0].clientX)
+    setCursorOn(false)
+  }
+
+  useEffect(() => {
+    const dragSpace = Math.abs(startClientX - endClientX)
+    const userSlideRight = endClientX < startClientX && dragSpace > 100
+    const userSlideLeft = endClientX > startClientX && dragSpace > 100
+    if (startClientX === 0) {
+      return
+    }
+    if (userSlideRight) {
+      handleOffset('RIGHT')
+    } else if (userSlideLeft) {
+      handleOffset('LEFT')
+    }
+  }, [endClientX])
+
+  return (
+    <StyledCarouselWrapper>
+      <StyledSlider
+        cursorOn={cursorOn}
+        size={carouselWidthSize}
+        onMouseDown={handleMouseDown}
+        onMouseUp={handleMouseUp}
+        onTouchEnd={handleTouchStartEnd}
+        onTouchStart={handleTouchStart}>
+        <StyledImageBox translateValue={translateValue || null}>
+          {images.map(image => {
+            return (
+              <StyledImage
+                key={image.id}
+                alt={`${name}- ${image.id}`}
+                size={carouselWidthSize}
+                src={image.url}
+              />
+            )
+          })}
+        </StyledImageBox>
+        {isArrow && (
+          <StyledArrowBox>
+            <StyledArrow
+              alt="arrow-left"
+              src={arrowLeft}
+              onClick={(): void => {
+                handleOffset('LEFT')
+              }}
+            />
+            <StyledArrow
+              alt="arrow-right"
+              src={arrowRight}
+              onClick={(): void => {
+                handleOffset('RIGHT')
+              }}
+            />
+          </StyledArrowBox>
+        )}
+      </StyledSlider>
+      <StyledIndicatorBox>
+        {images.map(image => {
+          return (
+            <StyledIndicator
+              key={image.id}
+              className={`${name}- ${image.id}`}
+              onClick={(): void => {
+                handleIndicator(image.id)
+              }}></StyledIndicator>
+          )
+        })}
+        <StyledCurrentIndicator
+          imageIndex={
+            translateValue / carouselWidthSize
+          }></StyledCurrentIndicator>
```

Comment:
```suggestion
          }/>
```
요기도 single 반영이 안된 것 같아용!